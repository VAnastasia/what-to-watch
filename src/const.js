const MAX_COUNT_GENRES = 9;
const GENRE_DEFAULT = `All genres`;

const START_SHOWN_MOVIES_AMOUNT = 8;
const SHOW_MOVIES_ON_CLICK = 8;

const TabName = {
  OVERVIEW: `Overview`,
  DETAILS: `Details`,
  REVIEWS: `Reviews`,
};

export {MAX_COUNT_GENRES, GENRE_DEFAULT, TabName, START_SHOWN_MOVIES_AMOUNT, SHOW_MOVIES_ON_CLICK};
