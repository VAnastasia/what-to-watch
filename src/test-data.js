const promoFilm = {
  title: `The Grand Budapest Hotel`,
  genre: `Drama`,
  year: 2014
};

const films = [
  {
    id: 1,
    title: `Fantastic Beasts: The Crimes of Grindelwald`,
    image: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    video: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  },
  {
    id: 2,
    title: `Bohemian Rhapsody`,
    image: `img/bohemian-rhapsody.jpg`,
    video: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  },
  {
    id: 3,
    title: `Macbeth`,
    image: `img/macbeth.jpg`,
    video: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  },
  {
    id: 4,
    title: `Aviator`,
    image: `img/aviator.jpg`,
    video: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  },
  {
    id: 5,
    title: `We need to talk about Kevin`,
    image: `img/we-need-to-talk-about-kevin.jpg`,
    video: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  }
];

const film = {
  id: 1,
  title: `The Grand Budapest Hotel`,
  image: `img/bg-the-grand-budapest-hotel.jpg`,
  video: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  poster: `img/the-grand-budapest-hotel-poster.jpg`,
  genre: `drama`,
  year: `2014`,
  rating: 8.9,
  ratingCount: 240,
  description: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege. Gustave prides himself on providing first-class service to the hotel's guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustave's lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.`,
  director: `Wes Andreson`,
  actors: [
    `Bill Murray`,
    `Edward Norton`,
    `Jude Law`,
    `Willem Dafoe`
  ]
};

export {promoFilm, films, film};
