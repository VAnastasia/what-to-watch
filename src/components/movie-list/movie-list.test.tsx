import * as React from "react";
import * as renderer from "react-test-renderer";
import MovieList from "./movie-list";
import {films} from "../../test-data";
import {noop} from "../../const";

it(`Should MovieList render correctly`, () => {
  const tree = renderer
    .create(<MovieList
      movies={films}
      loadComments={noop}
    />,
    {
      createNodeMock: () => {
        return {};
      }
    })
    .toJSON();

  expect(tree).toMatchSnapshot();
});
