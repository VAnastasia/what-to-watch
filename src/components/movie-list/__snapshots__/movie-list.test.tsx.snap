// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should MovieList render correctly 1`] = `
<div
  className="catalog__movies-list"
>
  <article
    className="small-movie-card catalog__movies-card"
    onClick={[Function]}
    onMouseLeave={[Function]}
    onMouseOver={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <video
        autoPlay={true}
        height="175"
        muted={true}
        poster="https://htmlacademy-react-3.appspot.com/wtw/static/film/preview/snatch.jpg"
        src=""
        width="280"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
      >
        Snatch
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onClick={[Function]}
    onMouseLeave={[Function]}
    onMouseOver={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <video
        autoPlay={true}
        height="175"
        muted={true}
        poster="https://htmlacademy-react-3.appspot.com/wtw/static/film/preview/snatch.jpg"
        src=""
        width="280"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
      >
        Snatch
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onClick={[Function]}
    onMouseLeave={[Function]}
    onMouseOver={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <video
        autoPlay={true}
        height="175"
        muted={true}
        poster="https://htmlacademy-react-3.appspot.com/wtw/static/film/preview/snatch.jpg"
        src=""
        width="280"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
      >
        Snatch
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onClick={[Function]}
    onMouseLeave={[Function]}
    onMouseOver={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <video
        autoPlay={true}
        height="175"
        muted={true}
        poster="https://htmlacademy-react-3.appspot.com/wtw/static/film/preview/snatch.jpg"
        src=""
        width="280"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
      >
        Snatch
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onClick={[Function]}
    onMouseLeave={[Function]}
    onMouseOver={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <video
        autoPlay={true}
        height="175"
        muted={true}
        poster="https://htmlacademy-react-3.appspot.com/wtw/static/film/preview/snatch.jpg"
        src=""
        width="280"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
      >
        Snatch
      </a>
    </h3>
  </article>
</div>
`;
